src/clr-core/common/base/button.base.ts

```diff
@@ -72,10 +83,55 @@ export class CwcBaseButton extends LitElement {
     }
   }
 
+  /**
+   * We have to append a hidden button outside the web component in the light DOM
+   * This allows us to trigger native submit events within a form element.
+   */
+  private setHiddenButton() {
+    if (!this.readonly) {
+      this.hiddenButton = this.appendChild(this.templateButton);
+    }
+  }
+
+  private setEventListeners() {
+    if (!this.readonly) {
```

Comment:
if i toggle the `readonly` value from, say, `false` to `true`, does it go back and remove these event listeners? or do the event listeners still stick around and fire?

Reply:
oh good catch! It should check on each event if its still in readonly or not. If I remember correctly the events will also be cleaned up when the component is destroyed automatically.

Reply:
Simplified logic to check for updates